src/input.ts

```diff
@@ -246,3 +216,64 @@ export class Mouse {
     await this.click(x, y, { ...options, clickCount: 2 });
   }
 }
+
+const aliases = new Map<string, string[]>([
+  ['ShiftLeft', ['Shift']],
+  ['ControlLeft', ['Control']],
+  ['AltLeft', ['Alt']],
+  ['MetaLeft', ['Meta']],
+  ['Enter', ['\n', '\r']],
+]);
+
+const qwertyLayout = buildLayoutClosure(keyboardLayout.Qwerty);
+
+function buildLayoutClosure(layout: keyboardLayout.KeyboardLayout): Map<string, KeyDescription> {
+  const result = new Map<string, KeyDescription>();
+  for (const code in layout) {
+    const definition = layout[code];
+    const description: KeyDescription = {
+      key: definition.key || '',
+      keyCode: definition.keyCode || 0,
+      keyCodeWithoutLocation: definition.keyCodeWithoutLocation || definition.keyCode || 0,
+      code,
+      text: definition.text || '',
+      location: definition.location || 0,
+    };
+    if (definition.key.length === 1)
+      description.text = description.key;
+
+    // Generate shifted definition.
+    let shiftedDescription: KeyDescription | undefined;
+    if (definition.shiftKey) {
+      assert(definition.shiftKey.length === 1);
+      shiftedDescription = { ...description };
+      shiftedDescription.key = definition.shiftKey;
+      shiftedDescription.text = definition.shiftKey;
+      if (definition.shiftKeyCode)
+        shiftedDescription.keyCode = definition.shiftKeyCode;
+      description.shifted = shiftedDescription;
+    }
+
+    // Map from code: Digit3 -> { ... descrption, shifted }
+    result.set(code, { ...description, shifted: shiftedDescription });
+
+    // Map from aliases: Shift -> non-shiftable definition
+    if (aliases.has(code)) {
+      for (const alias of aliases.get(code)!)
+        result.set(alias, description);
+    }
+
+    // Do not use numpad when converting keys to codes.
+    if (definition.location)
```

Comment:
This can be removed if order matters in usKeyboardLayout

Reply:
Yeah, but it becomes subtle that way. I'd rather use a bigger hammer.